src/main/java/org/zalando/nakadi/controller/CursorsController.java

```diff
@@ -31,44 +39,85 @@
+                             final EventTypeRepository eventTypeRepository) {
         this.cursorsService = cursorsService;
         this.featureToggleService = featureToggleService;
+        this.subscriptionRepository = subscriptionRepository;
+        this.eventTypeRepository = eventTypeRepository;
     }
 
     @RequestMapping(path = "/subscriptions/{subscriptionId}/cursors", method = RequestMethod.GET)
     public ResponseEntity<?> getCursors(@PathVariable("subscriptionId") final String subscriptionId,
-                                        final NativeWebRequest request) {
+                                        final NativeWebRequest request,
+                                        final Client client) {
         if (!featureToggleService.isFeatureEnabled(HIGH_LEVEL_API)) {
             return new ResponseEntity<>(NOT_IMPLEMENTED);
         }
-
         try {
+            validateSubscriptionReadScopes(client, subscriptionId);
             final List<SubscriptionCursor> cursors = cursorsService.getSubscriptionCursors(subscriptionId);
-            return status(OK).body(cursors);
+            return status(OK).body(new ItemsWrapper<>(cursors));
         } catch (final NakadiException e) {
             return create(e.asProblem(), request);
         }
     }
 
-    @RequestMapping(value = "/subscriptions/{subscriptionId}/cursors", method = RequestMethod.PUT)
+    @RequestMapping(value = "/subscriptions/{subscriptionId}/cursors", method = RequestMethod.POST)
     public ResponseEntity<?> commitCursors(@PathVariable("subscriptionId") final String subscriptionId,
                                            @RequestBody final List<SubscriptionCursor> cursors,
-                                           final NativeWebRequest request) {
+                                           final NativeWebRequest request,
+                                           final Client client) {
         if (!featureToggleService.isFeatureEnabled(HIGH_LEVEL_API)) {
             return new ResponseEntity<>(NOT_IMPLEMENTED);
         }
 
         try {
-            final boolean allCommitted = cursorsService.commitCursors(subscriptionId, cursors);
-            return allCommitted ? ok().build() : noContent().build();
+            validateSubscriptionReadScopes(client, subscriptionId);
+            final Map<SubscriptionCursor, Boolean> result = cursorsService.commitCursors(subscriptionId, cursors);
+            final List<CursorCommitResult> items = result.entrySet().stream()
+                    .map(entry -> new CursorCommitResult(entry.getKey(), entry.getValue()))
+                    .collect(Collectors.toList());
+            final boolean allCommited = result.values().stream().reduce(Boolean::logicalAnd).orElseGet(() -> true);
+            final ItemsWrapper<CursorCommitResult> body = new ItemsWrapper<>(items);
+            return allCommited ? status(201).body(body) : ok(body);
```

Comment:
Other two questions
1) Why it's 201? In API it's 204.
2) In a case when all cursors are committed the body should be empty.